### PR TITLE
SubS ActorPathing

### PR DESCRIPTION
**Pull request by hensldm**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Part 6 of SubS:

This just includes a general system for handling actors that follow some path (https://discordapp.com/channels/688807550715560050/873211000122384415/945200696842674246).

Only used by 4 actors as far as I can tell.
 * EnOt (Seahorse)
 * EnDno (Deku Butler)
 * EnAz (Beavers)
 * BgDanpeiMovebg (Moving platforms beneath Dampe's house)

### CODE REVIEW

File: src/overlays/actors/ovl_En_Dno/z_en_dno.c [hunk 1]
```diff
@@ -788,11 +788,12 @@ void func_80A730A0(EnDno* this, GlobalContext* globalCtx) {
         }
     }
 
-    func_8013DE04(globalCtx, &this->unk_340, func_8013DF3C, func_80A72D8C, func_80A72FAC, func_8013E0A4);
+    SubS_ActorPathing_Update(globalCtx, &this->actorPath, SubS_ActorPathing_ComputePointInfo, func_80A72D8C,
+                             func_80A72FAC, SubS_ActorPathing_SetNextPoint);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Dno/z_en_dno.c:792]: What do you think about renaming those callbacks too?

File: src/overlays/actors/ovl_En_Ot/z_en_ot.c [hunk 2]
```diff
@@ -900,16 +895,17 @@ void func_80B5D648(EnOt* this, GlobalContext* globalCtx) {
 
 void func_80B5D750(EnOt* this, GlobalContext* globalCtx) {
     if (!(this->unk_32C & 1) && !(this->unk_32C & 2)) {
-        func_8013DE04(globalCtx, &this->unk_2C0, func_8013DF3C, func_80B5D470, func_80B5D37C, func_8013E0A4);
+        SubS_ActorPathing_Update(globalCtx, &this->actorPath, SubS_ActorPathing_ComputePointInfo, func_80B5D470,
+                                 func_80B5D37C, SubS_ActorPathing_SetNextPoint);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Ot/z_en_ot.c:899]: Same here

File: include/z64.h [hunk 3]
```diff
@@ -1392,33 +1392,48 @@ typedef s32 (*func_8013E748_arg6)(struct GlobalContext*, Actor*, Vec3s*);
...
-    /* 0x68 */ struct_8013DF3C_arg1_unk_func2 unk_68;
-} struct_8013DF3C_arg1; // size = 0x6C
+    /* 0x4C */ f32 distSqToCurPointXZ;
+    /* 0x50 */ f32 distSqToCurPoint;
+    /* 0x54 */ Vec3s rotToCurPoint;
+    /* 0x5C */ ActorPathingComputeFunc computePointInfoFunc;
+    /* 0x60 */ ActorPathingUpdateFunc updateActorInfoFunc; // Return true if should setNextPoint, false if the actor should move forward
+    /* 0x64 */ ActorPathingUpdateFunc moveFunc; // Return true if should compute and update again
+    /* 0x68 */ ActorPathingUpdateFunc setNextPointFunc; // Return true if should compute and update again
+} ActorPathing; // size = 0x6C
```
[the remark this answers is not in the document]
Comment (hensldm) [reply include/z64.h]: Done, did it in oneish commit so can easily revert if anyone has objections.

**Review by AngheloAlf — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]
this is the way

**Review by Kenix3 — APPROVED** [review]
Good documentation.

**Pull request merged**